coil-compose-base/src/main/java/coil/compose/AsyncImagePainter.kt

```diff
@@ -247,17 +251,19 @@ class AsyncImagePainter internal constructor(
     internal var contentScale = ContentScale.Fit
     internal var filterQuality = DefaultFilterQuality
     internal var isPreview = false
+    internal val requestInternal = MutableStateFlow(request)
 
     /** The current [AsyncImagePainter.State]. */
     var state: State by mutableStateOf(State.Empty)
         private set
 
     /** The current [ImageRequest]. */
-    var request: ImageRequest by mutableStateOf(request)
-        internal set
+    val request
+        @Composable
+        get() = requestInternal.collectAsState().value
```

Comment:
@colinrtwhite what do you think about this change? Technically this changes the API, because now the `request` is only available from a composable, which wasn't before. I'm not sure how this could be done in a backwards compatible way 😬

Reply:
I'd prefer to keep it as a non-Composable since it's then the same as the other exposed properties (`imageLoader` and `state`). Though if there's a strong performance case then we should consider changing all the exposed properties to be `@Composable`.

Reply:
@mlykotom I think 0.2ms is probably a strong enough incentive to change the API. Though unfortunately, we can't make any breaking changes to the 2.x branch. We could make the changes on the 3.x (`main`) branch, however.